APIMVI/ViewController.swift

```diff
@@ -7,45 +7,97 @@
 //
 
 import UIKit
-import Foundation
 import RxSwift
 import RxCocoa
-import RxAlamofire
-import Alamofire
 
 class ViewController: UIViewController {
+  private let disposeBag = DisposeBag()
 
-  let disposeBag = DisposeBag()
+  private let lifecycleRelay = PublishRelay<MviLifecycle>()
+  private let statesRelay = BehaviorRelay<BlogState?>(value: nil)
+  private let networkManager = NetworkManager()
 
+  // Intentions
+  private lazy var retryIntention = retryButton.rx.tap.asObservable()
+  private lazy var searchIntention = searchTextField.rx.text.orEmpty.asObservable()
+  private lazy var intentions = BlogIntentions(retryIntention, searchIntention)
+
+  // UI Components
+  @IBOutlet private weak var retryButton: UIButton!
+  @IBOutlet private weak var searchTextField: UITextField!
+  @IBOutlet private weak var tableView: UITableView!
+  
+  @IBOutlet weak var activityIndicatorView: UIActivityIndicatorView! {
+    didSet {
+      let scale: CGFloat = 3
+      activityIndicatorView.transform = CGAffineTransform(scaleX: scale, y: scale)
+    }
+  }
+  
   override func viewDidLoad() {
     super.viewDidLoad()
 
-    getRequest().subscribe(onNext: { (arr) in
-      print("Array Count")
-      print(arr.count)
-    }, onError: {
-      (error) in
-      print("Error")
-      print(error.localizedDescription)
-    }).disposed(by: disposeBag)
+    setup()
+    lifecycleRelay.accept(.created)
+  }
+
+  func setup() {
+    let statesObservable = statesRelay
+      .filter { return $0 != nil }
+      .map { $0! }
+
+    bind(lifecycleRelay.asObservable(), networkManager, intentions, statesObservable)
+  }
+
+  private func bind(
+    _ lifecycle: Observable<MviLifecycle>,
+    _ networkManager: NetworkManager,
+    _ intentions: BlogIntentions,
+    _ states: Observable<BlogState>
+  ) {
+    BlogsModel
+      .bind(lifecycle, networkManager, intentions, states.asObservable())
+      .observeOn(MainScheduler.asyncInstance)
+      .subscribeOn(MainScheduler.asyncInstance)
+      .subscribe { (state) in
+        if let state = state.element {
+          self.statesRelay.accept(state)
+          self.render(state: state)
+        }
+      }
+      .disposed(by: disposeBag)
   }
+}
+
+extension ViewController: BlogsView {
+  func showLoading(show: Bool) {
+    if show {
+      activityIndicatorView.startAnimating()
+    } else {
+      activityIndicatorView.stopAnimating()
+    }
+  }
+
+  func showRetry(show: Bool) {
+    retryButton.isHidden = !show
+    tableView.isHidden = show
+    searchTextField.isHidden = show
+  }
+
+  func showBlogs(blogs: [Blog]) {
+    let blogs = Observable.of(
```

Comment:
Any reason why this is an `Observable`?

Reply:
I think it's because they're using rx table view. @kapilrathore this might not be needed.